src/VisualStudio/Core/Def/ValueTracking/ValueTrackedTreeItemViewModel.cs

```diff
@@ -82,28 +76,35 @@ private void CalculateChildren()
                 return;
             }
 
+            TreeViewModel.LoadingCount++;
             _childrenCalculated = true;
 
-            CalculateChildrenAsync(CancellationToken.None)
-                 .ContinueWith(task =>
-                 {
-                     if (task.Exception is not null)
-                     {
-                         _childrenCalculated = false;
-                         return;
-                     }
-
-                     ChildItems.Clear();
-
-                     foreach (var item in task.Result)
-                     {
-                         ChildItems.Add(item);
-                     }
-                 },
-
-                 // Use the UI thread synchronization context for calling back to the UI
-                 // thread to add the tiems
-                 TaskScheduler.FromCurrentSynchronizationContext());
+            System.Threading.Tasks.Task.Run(async () =>
+            {
+                try
+                {
+                    var items = await _valueTrackingService.TrackValueSourceAsync(_trackedItem, ThreadingContext.DisposalToken).ConfigureAwait(false);
+
+                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    ChildItems.Clear();
```

Comment:
Since you're clearing the child items before adding here anyway, is it easy to add a child "Loading..." item while things are processing?

Obviously this is not necessary if a spinner is possible.

Reply:
I think clearing the items first and then changing the "icon" to loading will look good.  Similar to what @Cosifne said in https://github.com/dotnet/roslyn/pull/53206#issuecomment-833208265

Reply:
I think the look @Cosifne suggested with https://github.com/dotnet/roslyn/pull/53206#issuecomment-833208265 will be easy to do, and can clear out children to remove the empty space there

Reply:
![bbae2651-693a-4319-9794-139267b9582b](https://user-images.githubusercontent.com/475144/117365578-588ce480-ae74-11eb-9420-c726c74b910d.gif)